numpy/core/_asarray.py

```diff
@@ -181,11 +219,22 @@ def ascontiguousarray(a, dtype=None):
     so it will not preserve 0-d arrays.  
 
     """
+    if like is not None:
+        return _ascontiguousarray_with_like(a, dtype=dtype, like=like)
+
     return array(a, dtype, copy=False, order='C', ndmin=1)
 
 
+@array_function_dispatch(
+    _asarray_contiguous_fortran_dispatcher,
+    public_api=ascontiguousarray,
+)
+def _ascontiguousarray_with_like(a, dtype=None, *, like=None):
+    pass
```

Comment:
This strikes me as really weird. The following spelling would also work, and avoids repeating the signature:
```
_ascontiguousarray_with_like = array_function_dispatch(
    _asarray_contiguous_fortran_dispatcher
)(ascontiguousarray)
```

Reply:
Thanks, I definitely missed we could do something simple like this. Done now in https://github.com/numpy/numpy/pull/16935/commits/d83cf9d0ea714f291e001f653ef0c8753f689248 .